src/SimulationService.jl

```diff
@@ -1,77 +1,352 @@
+    timespan::Tuple{Float64, Float64}
+    df::DataFrame                       # empty if !haskey(obj, :dataset)
+    job_id::String                      # auto-generated in constructor
+    operation::String                   # e.g. "simulate"
+
+    function OperationRequest(req::HTTP.Request)
+        obj = JSON3.read(req.body)
+        model_config_ids = get_model_config_ids(obj)
+        amr = map(id -> get_json("$SIMSERVICE_TDS_URL/model_configurations/$id"), model_config_ids)
+        timespan = get_timespan(obj)
+        df = get_df(obj)
+        job_id = "sciml-$(UUIDs.uuid4())"
+        operation = split(req.url.path, ",")[end]
+
+        new(obj, model_config_ids, amr, timespan, df, job_id, operation)
+    end
+end
+
+function get_model_config_ids(obj::JSON3.Object)
+    if haskey(obj, :model_config_id)
+        return [UUID(obj.model_config_id)]
+   elseif haskey(obj, :model_config_ids)
+        return UUID.(obj.model_config_ids)
+   else
+       error("JSON request to have a `model_config_id` or `model_config_ids` key.")
+   end
+end
+
+function get_timespan(obj::JSON3.Object)
+    if haskey(obj, :timespan)
+        map(float, (obj.timespan.start, obj.timespan.end))
+    else
+        @warn "JSON request doesn't contain `timespan`.  Setting to (0.0, 100.0)."
+        (0.0, 100.0)
+    end
+end
+
+function get_df(obj::JSON3.Object)
+    !haskey(obj, :dataset) && return DataFrame()
+    (; dataset_id, filename) = obj.dataset
+    data_url = "$SIMSERVICE_TDS_URL/datasets/$dataset_id/download-url?filename=$filename"
+    df = CSV.read(download(get_json(data_url).url), DataFrame)
+    rename!(df, Dict(obj.dataset.mappings))
+    return df
+end
+
+#-----------------------------------------------------------------------------# POST /operation/{operation}
+function operation(req::HTTP.Request, op::String)
+    try
+        # Get T <: Operation from request
+        T = get_operation_type(op)
+        op_req = OperationRequest(req)
+        op = T(op_req)
+        callback = get_callback(op_req.job_id)
+
+        # schedule job and update status when finished
+        tds_update_job(o.job_id, Dict(:status => "running", :start_time => time()))
+        t = @task function ()
+            output = solve(op; callback)
+            prepare_output(output, op_req)
+            tds_update_job(o.job_id, Dict(:status => "complete", :completed_time => time()))
+        end
+        sim_run = JobSchedulers.Job(t)
+        sim_run.id = o.job_id
+        JobSchedulers.submit!(sim_run)
+
+        body = JSON3.write((; simulation_id = o.job_id))
+        return Response(201, ["Content-Type" => "application/json; charset=utf-8"]; body=body)
+    catch ex
+        return Response(500, ["Content-Type" => "application/json; charset=utf-8"]; body=JSON3.write((; error=string(ex))))
+    end
+end
+
+function get_operation_type(op::String)
+    d = Dict(lowercase(string(T)) => T for T in subtypes(Operation))
```

Comment:
This should be a constant; `subtypes` is very slow and requires `InteractiveUtils`.